.github/workflows/test.yml

```diff
@@ -0,0 +1,158 @@
+# This GitHub workflow will setup and run various kinds of tests with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: test
+
+on:
+  # schedule:
+  # # minute hour day_of_month month day_of_week
+  # - cron:  '10 21 * * 0'  # Every sunday at 21:10 UTC
+  push:
+    branches: [ master, stable_1.1 ]
+  pull_request:
+    branches: [ master, stable_1.1 ]
+
+jobs:
+
+  test:
+    strategy:
+      fail-fast: false
+      max-parallel: 20
+      # # The matrix for the all-environment pull request:
+      # matrix:
+      #   os: [ubuntu-latest, macos-latest, windows-latest]
+      #   python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9]
+      #   package_level: [minimum, latest]
+      #   exclude:
+      #   - os: macos-latest
+      #     python-version: 3.4
+      #     package_level: minimum
+      #   - os: macos-latest
+      #     python-version: 3.4
+      #     package_level: latest
+      #   - os: windows-latest
+      #     python-version: 3.4
+      #     package_level: minimum
+      #   - os: windows-latest
+      #     python-version: 3.4
+      #     package_level: latest
+      # The matrix for normal pull requests:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [2.7, 3.4, 3.9]
+        package_level: [minimum, latest]
+        include:
+        - os: macos-latest
+          python-version: 3.9
+          package_level: latest
+        - os: windows-latest
+          python-version: 3.8
+          package_level: minimum
+        - os: windows-latest
```

Comment:
Is there a reason that this has separate definitionf with just package_level as difference instead of [package_level: [latest, minimum]

Reply:
Yes. The `include` statement only supports single values, and not lists, which would essentially cause them to add delta matrices.

Also, it requires specifying a value for each dimension (i.e. does not inherit it from defaults).